File: python/day_1.py
def num_times_increases(depths: list[int]) -> int:
    count = 0

    for i, depth in enumerate(depths):
        if i != 0 and depth > depths[i - 1]:
            count += 1

    return count


def num_times_increases_sliding_window(depths: list[int]) -> int:
    sliding_window_sums: list[int] = []
    increase_count = 0

    for i, depth in enumerate(depths):
        try:
            window = depths[i - 1], depth, depths[i + 1]
        except IndexError:
            continue
        else:
            this_window_sum = sum(window)

            if sliding_window_sums and this_window_sum > sliding_window_sums[-1]:
                increase_count += 1

            sliding_window_sums.append(this_window_sum)

    return increase_count

File: cpp/src/Day1.cpp
#include "Day1.hpp"

int numTimesIncreases(const std::vector<int>& depths) { 
    int count = 0;

    for (int i = 0; i < depths.size(); ++i) {
        const auto& depth = depths[i];

        if (i != 0 && depth > depths[i - 1]) {
            ++count;
        }
    }

    return count;
 }

 int numTimesIncreasesSlidingWindow(const std::vector<int>& depths) {
     std::vector<int> sums{};
     int count = 0;

     for (int i = 1; i < depths.size() - 1; ++i) {
         const auto sum = depths[i - 1] + depths[i] + depths[i + 1];

         if (sums.size() != 0 && sum > sums.back()) {
             ++count;
         }

         sums.push_back(sum);
     }

     return count;
 }
